fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeFileStats.java

```diff
@@ -0,0 +1,302 @@
+import com.starrocks.sql.optimizer.statistics.ColumnStatistic;
+import io.delta.kernel.types.BasePrimitiveType;
+import io.delta.kernel.types.BooleanType;
+import io.delta.kernel.types.ByteType;
+import io.delta.kernel.types.DataType;
+import io.delta.kernel.types.DateType;
+import io.delta.kernel.types.DoubleType;
+import io.delta.kernel.types.FloatType;
+import io.delta.kernel.types.IntegerType;
+import io.delta.kernel.types.LongType;
+import io.delta.kernel.types.ShortType;
+import io.delta.kernel.types.StructField;
+import io.delta.kernel.types.StructType;
+import io.delta.kernel.types.TimestampNTZType;
+import io.delta.kernel.types.TimestampType;
+
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class DeltaLakeFileStats {
+    private static final DateTimeFormatter TIMESTAMP_NTZ_FORMAT =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS");
+    private static final DateTimeFormatter TIMESTAMP_ZONE_FORMAT =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+
+    private final StructType schema;
+    private final List<String> nonPartitionPrimitiveColumns;
+    private long recordCount;
+    private long size;
+    private final Map<String, Object> minValues;
+    private final Map<String, Object> maxValues;
+    private final Map<String, Object> nullCounts;
+    private final Set<String> corruptedStats;
+    private boolean hasValidColumnMetrics;
+
+    public DeltaLakeFileStats(StructType schema,
+                              List<String> nonPartitionPrimitiveColumns,
+                              long recordCount,
+                              long size,
+                              Map<String, Object> minValues,
+                              Map<String, Object> maxValues,
+                              Map<String, Object> nullCounts) {
+        this.schema = schema;
+        this.nonPartitionPrimitiveColumns = nonPartitionPrimitiveColumns;
+        this.recordCount = recordCount;
+        this.size = size;
+        if (minValues == null || maxValues == null || nullCounts == null) {
+            this.minValues = null;
+            this.maxValues = null;
+            this.nullCounts = null;
+            this.corruptedStats = null;
+            this.hasValidColumnMetrics = false;
+        } else {
+            this.minValues = minValues;
+            this.maxValues = maxValues;
+            this.nullCounts = nullCounts;
+            this.corruptedStats = nonPartitionPrimitiveColumns.stream()
+                    .filter(col -> !minValues.containsKey(col) &&
+                            (!nullCounts.containsKey(col) || ((Double) nullCounts.get(col)).longValue() != recordCount))
```

Comment:
why need to check nullCounts? if minValues do not contains the col, is it corrupted stats?

Reply:
For boolean type, there is no min/max values, but have null counts.

This code is copyed from iceberge catalog, i don't know why...